### PR TITLE
DEP deprecate using Travis CI usage on win-*, osx-* or linux_64

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

due to how travis CI now counts CI minutes, we can no longer use it easily on windows, osx or linux-64. It can be used easily on pcc64le, aarch64 and s390x.

### CODE REVIEW

**Comment by beckermr** [comment]
Thoughts @conda-forge/core @isuruf?

**Comment by beckermr** [comment]
See https://docs.travis-ci.com/user/billing-overview/#partner-queue-solution

**Review by isuruf — COMMENTED** [review]
Can you check that this is for conda-forge and only disable it then?
(We do the same for github_actions)

**Comment by isuruf** [comment]
https://github.com/conda-forge/conda-smithy/blob/c299172b8e86f16b94d5f02234e89df35c5440db/conda_smithy/configure_feedstock.py#L641-L650

File: news/travis.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+**Deprecated:**
+
+* We have deprecated the usage of Travis CI for any platforms but linux_aarch64, linux_ppc64le, or
+  linux_s390x. Conda-smithy will no longer render CI scripts for other platforms on travis and will
+  emit a warning if one attempts to do so.
```
Comment (beckermr) [inline news/travis.rst:5]:
```suggestion
  linux_s390x. Conda-smithy will raise a RuntimeError if one attempts to render a recipe for a different platform.
```

**Comment by beckermr** [comment]
ok @isuruf this is ready for another look!

**Review by isuruf — APPROVED** [review]

**Pull request merged**